### PR TITLE
CSL-1043 Unexpected block/net logic responses handling

### PR DESCRIPTION
**Pull request by volhovM**
See commit descriptions. This should stop node from infinitely asking something when receiving unexpected message (`Nothing` in `requestHeaders` in our case).

### CODE REVIEW

File: scripts/analyze/blocks.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-for f in node*.log; do
+for f in node*.log*; do
```
Comment (gromakovsky) [inline scripts/analyze/blocks.sh]: Want there be duplicates because of presence on `.pub` logs?

Comment (volhovM) [inline scripts/analyze/blocks.sh]: Hm, you're right. Let me fix that.

File: src/Pos/Block/Network/Logic.hs [hunk 2]
```diff
@@ -64,6 +68,27 @@ import           Pos.Util                   (inAssertMode, _neHead, _neLast)
...
+data BlockNetLogicException
+    = VerifyBlocksException Text
+      -- ^ Failed to verify blocks coming from node.
+    | DialogUnexpected Text
+      -- ^ Node's response in any network/block related logic was
+      -- unexpected.
+    deriving (Show)
+
+instance B.Buildable BlockNetLogicException where
+    build e = bprint ("BlockNetLogicException: "%shown) e
```
Comment (gromakovsky) [inline src/Pos/Block/Network/Logic.hs]: I find it strange.

Comment (volhovM) [inline src/Pos/Block/Network/Logic.hs]: `Buildable` instance is required to derive from `CardanoException`, but i didn't want to provide anything too complex.

**Pull request merged**